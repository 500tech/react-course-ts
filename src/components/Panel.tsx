import * as React from 'react';
import Rating from './common/Rating';
// import {fetchMovieDetails} from "../queries/movies";
// import api from '../utils/api';

interface ComponentProps {
  movie: {
    label: string,
    rating: number,
    id: number,
    description: string
  },
  updateRating: (id, val) => void,
  clearSelected: () => void
}

const Panel = (props: ComponentProps) => {
  if (!props.movie) {
    return null;
  }

  return (
    <div className="panel">
      <div className="panel-title">{props.movie.label}</div>
      <div className="panel-description">
        {props.movie.description}
      </div>

      <Rating
        amount={props.movie.rating}
        updateRating={val => props.updateRating(props.movie.id, val)} />

      <div
        className="panel-close"
        onClick={props.clearSelected}>
        close panel
      </div>
    </div>
  );
};

export default Panel;
