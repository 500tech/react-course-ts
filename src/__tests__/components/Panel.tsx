import * as React from 'react';
import { shallow } from 'enzyme';
import Panel from "../../components/Panel";
import { noop } from 'lodash';

const setup = ({ movie = null, updateRating = noop, clearSelected = noop } = {}) => ({
  movie,
  updateRating,
  clearSelected
});

describe('<Panel/>', () => {

  const movie = {
    label: 'test movie',
    rating: 1,
    id: 1,
    description: "description"
  };

  test('should render nothing when no movie is provided', () => {
    const props = setup();
    const component = shallow(
        <Panel {...props} />
    );

    expect(component).toMatchSnapshot();
  });

  test('should render a movie when a movie is provided', () => {
    const props = setup({ movie });

    const component = shallow(
        <Panel {...props} />
    );

    expect(component).toMatchSnapshot();
  });

  test('should call clearSelected when closing the panel', () => {
    const props = setup({ movie, clearSelected: jest.fn() });
    const component = shallow(
        <Panel {...props} />
    );

    expect(props.clearSelected).not.toHaveBeenCalled();

    component.find('.panel-close').simulate('click');

    expect(props.clearSelected).toHaveBeenCalled();
  });
});
