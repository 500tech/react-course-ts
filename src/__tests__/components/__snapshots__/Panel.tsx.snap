// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Panel/> should render a movie when a movie is provided 1`] = `
<div
  className="panel"
>
  <div
    className="panel-title"
  >
    test movie
  </div>
  <div
    className="panel-description"
  >
    description
  </div>
  <Rating
    amount={1}
    updateRating={[Function]}
  />
  <div
    className="panel-close"
    onClick={[Function]}
  >
    close panel
  </div>
</div>
`;

exports[`<Panel/> should render nothing when no movie is provided 1`] = `""`;
